pakyow-presenter/lib/pakyow/presenter/front_matter_parser.rb

```diff
@@ -0,0 +1,36 @@
+module Pakyow
+  module Presenter
+    # Module for parsing front matter from HTML
+    module FrontMatterParser
+
+      MATTER_MATCHER = /\A---\s*\n(.*?\n?)^---\s*$\n?/m
+
+      # Parses HTML and returns a hash of front matter info
+      #
+      def self.parse(html_string)
+        if match = html_string.match(MATTER_MATCHER)
+          info = YAML.load(match.captures[0])
+        end
+
+        # ensure info is a hash
```

Comment:
After making this change I realized we could simply return early at the top of the method instead of wrapping the whole thing in an if/else, so I've done that instead. That work for you?